### PR TITLE
CB-10689: Extend the data lake backup flow to trigger backup as well.

### PR DESCRIPTION
**Pull request by kkalvagadda1**
Until now data lake backup flow just triggers the database backup and then monitors the completion of data lake backup.

With the code changes made, data lake backup flow would do below.
1. Trigger data lake backup.
2. Trigger the database backup.
3. Monitors the completion of data lake backup

There are two init events for the data lake flow.
1. DATALAKE_DATABASE_BACKUP_EVENT
2. DATALAKE_TRIGGER_BACKUP_EVENT 

When the backup is triggered from CDP_CLI, DATALAKE_DATABASE_BACKUP_EVENT would be the flow initiating event.
When the backup is triggered from datalake-api, DATALAKE_TRIGGER_BACKUP_EVENT would be the flow initiating event.

**Testing Done:**
I have tested both the failure and success scenarios by running CB and datalake-dr service locally.
**Here are some of the tests**
**When triggered from datalake-api.**
1. Successful backup 
2. Database backup failure and made sure of proper flow completion.
3. Data lake backup failure and made sure of proper flow completion.
4. When datalake-dr service is down and made sure of proper flow completion.

**When triggered from cdp-cli**
1. Successful backup

**Sequence diagram for backup (Current behavior):**
https://www.websequencediagrams.com/files/render?link=raogude90AgSMddQYEZH2klGg7ABmGKZ9IsbIbiAbeJ6nAqQmw1hgcMo276fvrxN

**Sequence diagram for backup (Proposed behavior):**
https://www.websequencediagrams.com/files/render?link=5LiBPkbAYs1fUAbIntXBxAeS8smRHjX6rVNp5Gy72Un4DhVNWax5JO7B36hjTbja

**Sequence diagram for backup from UI (Proposed behavior):**
https://www.websequencediagrams.com/files/render?link=6ASOyw2LiGAL9uZywbZJQ6Lq5OSO5FswfRSXQHxH4IC5dAZlu9u41jLWOMbPlzRv

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/event/DatalakeTrigggerBackupEvent.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.sequenceiq.datalake.flow.dr.backup.event;
+
+import com.sequenceiq.datalake.entity.operation.SdxOperationType;
+import com.sequenceiq.datalake.flow.dr.event.DatalakeDatabaseDrStartBaseEvent;
+
+public class DatalakeTrigggerBackupEvent extends DatalakeDatabaseDrStartBaseEvent {
```
Comment (sodre90) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/event/DatalakeTrigggerBackupEvent.java:6]: *trigger

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java [hunk 2]
```diff
@@ -55,7 +67,7 @@
                     .from(DATALAKE_DATABASE_BACKUP_IN_PROGRESS_STATE)
                     .to(DATALAKE_FULL_BACKUP_IN_PROGRESS_STATE)
                     .event(DATALAKE_FULL_BACKUP_IN_PROGRESS_EVENT)
-                    .failureState(DATALAKE_DATABASE_BACKUP_FAILED_STATE)
+                    .failureState(DATALAKE_FULL_BACKUP_IN_PROGRESS_STATE)
```
Comment (sodre90) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java:70]: if we do this then  .from(DATALAKE_DATABASE_BACKUP_FAILED_STATE) .to(DATALAKE_FULL_BACKUP_IN_PROGRESS_STATE)... will never run, do we want this? If yes then we should remove that "from-to" (line 83-87)

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupActions.java [hunk 3]
```diff
@@ -167,16 +213,17 @@ protected void doExecute(SdxContext context, SdxEvent payload, Map<Object, Objec
...
+                    sdxBackupRestoreService.updateDatabaseStatusEntry(operationId, SdxOperationStatus.SUCCEEDED, null);
                 }
                 sendEvent(context, DatalakeFullBackupWaitRequest.from(context, backupId));
             }
 
             @Override
             protected Object getFailurePayload(SdxEvent payload, Optional<SdxContext> flowContext, Exception ex) {
-                return DatalakeDatabaseBackupFailedEvent.from(payload, ex);
+                return DatalakeBackupFailedEvent.from(payload, ex);
+//                return DatalakeDatabaseBackupFailedEvent.from(payload, ex);
```
Comment (sodre90) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupActions.java:226]: can we remove the comment?

Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupActions.java:226]: will update.

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/event/DatalakeTrigggerBackupEvent.java (hunk 1 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/event/DatalakeTrigggerBackupEvent.java:6]: will update

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java (hunk 2 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java:70]: You are right.

Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java:70]: you are right. Wil fix it.

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupActions.java (hunk 3 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupActions.java]: addressed.

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java (hunk 2 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/DatalakeDatabaseBackupFlowConfig.java]: addressed.

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/event/DatalakeTrigggerBackupEvent.java (hunk 1 above)
Comment (kkalvagadda1) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/event/DatalakeTrigggerBackupEvent.java]: addressed.

**Review by sodre90 — APPROVED** [review]

**Review by sodre90 — CHANGES_REQUESTED** [review]
[36mswagger-validation_1  |[0m The swagger spec at "/json/datalake.json" is invalid against swagger specification 2.0. see errors :
[36mswagger-validation_1  |[0m - "backupDatalake" is defined 2 times
[36mswagger-validation_1  |[0m

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java [hunk 4]
```diff
@@ -183,6 +185,22 @@ SdxClusterResponse create(@ValidStackNameFormat @ValidStackNameLength @PathParam
...
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "backup the datalake ", produces = MediaType.APPLICATION_JSON, nickname = "backupDatalake")
+    SdxBackupResponse backupDatalakeByName(@PathParam("name") String name,
+            @QueryParam("backupLocation") String backupLocation,
+            @QueryParam("backupName") String backupName);
+
+    @POST
+    @Path("{name}/backupDatalakeStatus")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "backup status of the datalake ", produces = MediaType.APPLICATION_JSON, nickname = "backupDatalake")
```
Comment (sodre90) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java:199]: backupDatalakeStatus

Comment (kkalvagadda1) [reply datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java]: I see the nickname is backupDatalake for both the new api's added.

Comment (kkalvagadda1) [reply datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java]: addressed.

**Review by sodre90 — APPROVED** [review]

**Pull request merged**